File: algorithms/flatland-space-station/flatland_space_station_v1.py
def flatlandSpaceStation(n, c):
  if (n == len(c)):
    return 0
  
  cities = [x for x in range(n)]
  prevNearestSpaceStation = 0
  nearestSpaceStation = min(c)
  citiesWithoutSpaceStation = []
  tmp = []
  
  for i in range(n):
    if cities[i] not in c:
      citiesWithoutSpaceStation.append(cities[i])
      continue    

    prevNearestSpaceStation = nearestSpaceStation
    nearestSpaceStation = cities[i]
    
    for city in citiesWithoutSpaceStation:
      if (abs(city - prevNearestSpaceStation) <= nearestSpaceStation - city):
        tmp.append(abs(city - prevNearestSpaceStation))
      else:
        tmp.append(nearestSpaceStation - city) 
 
    citiesWithoutSpaceStation = []
  
  for city in citiesWithoutSpaceStation:
    tmp.append(city - nearestSpaceStation)
  
  return max(tmp)

File: algorithms/flatland-space-station/flatlandSpaceStationV1.php
<?php
function flatlandSpaceStation($n, $c) {
  if ($n == count($c)) {
      return 0;
  }

  $noSpaceStation = [];
  $prevNearestSpaceStation = min($c);
  $maxD = 0;

  for ($i = 0; $i < $n; $i++) {
    if (!(in_array($i, $c))) {
      array_push($noSpaceStation, $i);
      continue;
    }

    foreach ($noSpaceStation as $city) {
      $tmp = [abs($city - $prevNearestSpaceStation), abs($city - $i)];
      $nearestDistance = min($tmp);
      
      if ($nearestDistance > $maxD) {
        $maxD = $nearestDistance;
      }
    };

    $prevNearestSpaceStation = $i;
    $noSpaceStation = [];
  }

  return $maxD;
}
